### PR TITLE
created funtionality for deleting notes

### PR DESCRIPTION
**Pull request by culinovic-a**

### CODE REVIEW

File: src/app/notes/delete-modal/delete-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div mat-dialog-content>
+  <h4>Are You Sure You Want To Delete This Note</h4>
+</div>
+<div mat-dialog-actions class="buttons">
+  <button mat-button [mat-dialog-close]="false">Cancel</button>
+  <button mat-button [mat-dialog-close]="true" cdkFocusInitial>Delete</button>
```
Comment (tabuckner) [inline src/app/notes/delete-modal/delete-modal.component.html:6]:
```suggestion
  <button mat-raised-button color="primary" [mat-dialog-close]="true" cdkFocusInitial>Delete</button>
```

File: src/app/notes/notes-list-item/notes-list-item.component.html [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <mat-card class="example-card">
   <mat-card-header>
     <mat-card-title>{{ note.noteTitle }}</mat-card-title>
+    <mat-icon (click)="deleteNote(note)">delete</mat-icon>
```
Comment (tabuckner) [inline src/app/notes/notes-list-item/notes-list-item.component.html:4]:
```suggestion
    <button mat-icon-button (click)="deleteNote(note)">
      <mat-icon>delete</mat-icon>
    </button>
```

File: src/app/notes/notes-list-item/notes-list-item.component.scss [hunk 3]
```diff
@@ -7,4 +7,8 @@
     flex-direction: row;
     justify-content: space-between;
   }
+
```
Comment (tabuckner) [inline src/app/notes/notes-list-item/notes-list-item.component.scss:10]: remove these rules when using mat-icon-button :)

File: src/app/notes/notes.module.ts [hunk 4]
```diff
@@ -23,5 +29,6 @@ import { MatCardModule } from '@angular/material/card';
     MatTooltipModule,
     MatCardModule,
   ],
+  entryComponents: [DeleteModalComponent],
```
Comment (tabuckner) [inline src/app/notes/notes.module.ts:32]: good job working through this one :)

File: src/app/notes/notes-list-item/notes-list-item.component.ts [hunk 5]
```diff
@@ -8,7 +10,21 @@ import { Note } from '../../new-note/models/note';
...
+  ) {}
 
   ngOnInit() {}
+
+  deleteNote(note: Note) {
+    this.deleteDialogService
+      .openDialog()
+      .afterClosed()
+      .subscribe((res) => {
+        if (res) {
```
Comment (tabuckner) [inline src/app/notes/notes-list-item/notes-list-item.component.ts:25]: what is the type of res?
what do we use res for?

**Comment by culinovic-a** [comment]
> When are we going to address the layout?

I left the layout improvements for the next ticket - that's why I worked on the functionality only and left it like this

File: src/app/notes/notes-list-item/notes-list-item.component.ts (hunk 5 above)
Comment (culinovic-a) [reply src/app/notes/notes-list-item/notes-list-item.component.ts:25]: boolean which I am using for the response from observable that I'm getting with afterClosed() method - thanks to this I'm checking whether my boolean is true or false, which then leads to my next function (deleting that note or closing modal)

**Pull request merged**